rplugin/python3/nvim_ipy/__init__.py

```diff
@@ -296,18 +302,32 @@ def ignore(self, msg_id):
         self.handle(msg_id, None)
 
     @neovim.function("IPyConnect", sync=True)
-    def ipy_connect(self, args):
+    def ipy_connect(self, args=(), async=True):
         self.configure()
+
+        window = True
+        if '--no-window' in args:
+            args.remove('--no-window')
+            window = False
+
         # create buffer synchronously, as there is slight
         # racyness in seeing the correct current_buffer otherwise
-        self.create_outbuf()
+        self.create_outbuf(window)
         # 'connect' waits for kernelinfo, and so must be async
-        Async(self).connect(args)
+        # unless requested otherwise
+        if async:
+            Async(self).connect(args)
+        else:
+            self.connect(args)
 
     @neovim.function("IPyRun")
     def ipy_run(self, args):
         code = args[0]
         silent = bool(args[1]) if len(args) > 1 else False
+
+        if self.km is None:
```

Comment:
I would argue against this, it muddles the interface needlessly. But I would support a change that makes possible for an exernal mapping to do this. Maybe a variable with connection status, and `User` autocmd when connected would be useful?

Reply:
Maybe lets then just make a variable making this behavior optional? Sort of `g:nvim_ipy_autostart` with default = 0.